src/basics/custom.md

```diff
@@ -0,0 +1,493 @@
+
+use crate::external_messages::{ExternalMsg, ExternalQuery};
+
+pub struct CustomModule {
+    pub is_poked: Item<'static, bool>,
+}
+
+impl CustomModule {
+    pub fn new() -> Self {
+        Self {
+            is_poked: Item::new("is_poked"),
+        }
+    }
+
+    pub fn setup(&self, storage: &mut dyn Storage) -> StdResult<()> {
+        self.is_poked.save(storage, &true)
+    }
+}
+
+impl Module for CustomModule {
+    type ExecT = ExternalMsg;
+    type QueryT = ExternalQuery;
+    type SudoT = Empty;
+
+    fn execute<ExecC, QueryC>(
+        &self,
+        _api: &dyn Api,
+        storage: &mut dyn Storage,
+        _router: &dyn CosmosRouter<ExecC = ExecC, QueryC = QueryC>,
+        _block: &BlockInfo,
+        _sender: Addr,
+        msg: Self::ExecT,
+    ) -> anyhow::Result<AppResponse>
+    where
+        ExecC: Debug + Clone + PartialEq + JsonSchema + DeserializeOwned + 'static,
+        QueryC: CustomQuery + DeserializeOwned + 'static,
+    {
+        match msg {
+            ExternalMsg::Poke {} => {
+                self.is_poked.save(storage, &true)?;
+                Ok(AppResponse::default())
+            }
+        }
+    }
+
+    fn sudo<ExecC, QueryC>(
+        &self,
+        _api: &dyn Api,
+        _storage: &mut dyn Storage,
+        _router: &dyn CosmosRouter<ExecC = ExecC, QueryC = QueryC>,
+        _block: &BlockInfo,
+        _msg: Self::SudoT,
+    ) -> anyhow::Result<AppResponse>
+    where
+        ExecC: Debug + Clone + PartialEq + JsonSchema + DeserializeOwned + 'static,
+        QueryC: CustomQuery + DeserializeOwned + 'static,
+    {
+        Ok(AppResponse::default())
+    }
+
+    fn query(
+        &self,
+        _api: &dyn Api,
+        storage: &dyn Storage,
+        _querier: &dyn Querier,
+        _block: &BlockInfo,
+        request: Self::QueryT,
+    ) -> anyhow::Result<Binary> {
+        match request {
+            ExternalQuery::IsPoked {} => {
+                let is_poked = self.is_poked.load(storage)?;
+                to_binary(&is_poked).map_err(Into::into)
+            }
+        }
+    }
+}
+```
+
+With this module, we can move to testing.
+We are going to use [`BasicAppBuilder`](https://docs.rs/cw-multi-test/latest/cw_multi_test/type.BasicAppBuilder.html) to save ourselves from typing the all the generics.
+It's fine as we are only interested in setting `ExecC` and `QueryC` types.
+Installing our module is done via [`with_custom`](https://docs.rs/cw-multi-test/latest/cw_multi_test/type.BasicAppBuilder.html) method.
+In case you need to initialize some values in your module, you can use [`build`](https://docs.rs/cw-multi-test/latest/cw_multi_test/struct.AppBuilder.html#method.build) method.
+Generics are set on `mt_app`, and there is no need to set them on `App`.
+Running `poke` on our contract will send the `ExternalMsg::Poke`, which `App` will dispatch to our custom module.
+
+```rust
+use sylvia::multitest::App;
+
+use crate::contract::multitest_utils::CodeId;
```

Comment:
This appears suddenly here. It was not mentioned or explained before. I mean maybe in previous chapters, but for someone that jumps directly to this chapter this will be suprising. Maybe some link or explanation?

Reply:
Not sure what you mean.

`App` is explained in chapter regarding `MultiTest` support same as `CodeId`. This is no place to explain that as I would also have to explain basically everything. Every message generated and so on. This would heavily affect readability.

Reply:
As discussed offline `CodeId` is generated by `Sylvia` as part of `MultiTest` helpers. It is described in earlier chapter so no need to do that here.